common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/SignalizedIntersection.h

```diff
@@ -0,0 +1,138 @@
+ * the License.
+ */
+#include <lanelet2_core/primitives/RegulatoryElement.h>
+#include <boost/algorithm/string.hpp>
+#include <lanelet2_core/utility/Units.h>
+#include <lanelet2_extension/regulatory_elements/CarmaTrafficLight.h>
+#include <unordered_set>
+
+namespace lanelet
+{
+/**
+ * @brief Represents signalized intersection on the road. 
+ *
+ * SignalizedIntersection consists of ENTRY, EXIT, INTERIOR lanelets that are saved as parameters. 
+ * Although this class doesn't manage it, entry lanelets are expected to have CarmaTrafficSignal class to represent the traffic light.
+ * Therefore, this class can be understood as merely recording of which lanelets are part of the intersection, but traffic signal timer
+ * or which entry correlates to which exit information is handled by each CarmaTrafficSignal object itself.
+ *
+ * @ingroup RegulatoryElementPrimitives
+ * @ingroup Primitives
+ */
+enum class IntersectionSection {ENTRY, EXIT, INTERIOR};
+
+struct CarmaRoleNameString
+{
+  static constexpr char IntersectionEntry[] = "intersection_entry"; //saved and used internally as RoleName::Refers
+  static constexpr char IntersectionExit[] = "intersection_exit";  
+  static constexpr char IntersectionInterior[] = "intersection_interior";  
+};
+
+class SignalizedIntersection : public RegulatoryElement
+{
+public:
+  static constexpr char RuleName[] = "signalized_intersection";
+
+  /**
+   * @brief Returns the entry lanelets into the signalized intersection
+   *
+   * @return list of const lanelets
+   */
+  ConstLanelets getEntryLanelets() const;
+
+    /**
+   * @brief Returns the interior lanelets of signalized intersection
+   *
+   * @return list of const lanelets
+   */
+  ConstLanelets getInteriorLanelets() const;
+
+  /**
+   * @brief Returns the exit lanelets into the signalized intersection
+   *
+   * @return list of const lanelets
+   */
+  ConstLanelets getExitLanelets() const;
+
+  /**
+   * @brief Returns the carma traffic signals in the specified entry lanelet in the intersection
+   * @param entry lanelet in the intersection to get the traffic signals for
+   *
+   * @return vector of const traffic signals. Empty list if the lanelet is not an entry lanelet or not in the intersection
+   */
+  std::vector<CarmaTrafficLightConstPtr> getTrafficSignals(const ConstLanelet& llt) const;
+
+  /**
+   * @brief Returns the stop line of the specified lanelet in the intersection
+   *
+   * @return gets a const stop line for a lanelet, if there is one
+   */
+  Optional<lanelet::ConstLineString3d> getStopLine(const ConstLanelet& llt) const;
+
+  /**
+   * @brief Adds the lanelet into the specified section of the intersection
+   * @param lanelet to add  
+   * @param section section of the intersection to add to (entry, exit, interior)
+   * 
+   * NOTE: If entry lanelet is being added, the user must make sure that the regulatory element refers the lanelet back
+   *       For maps, laneletMapPtr->update(lanelet, regem) for full connection
```

Comment:
Could the method do this itself to remove the restriction?
For example, check if the added lanelet references this regulatory element. If it does not then add this element to that lanelet?

Reply:
yep I added that functionality